camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java

```diff
@@ -93,92 +93,140 @@ private static boolean validateParams(OperationParams params) {
         // only allow file resources (why this comment????)
         return true;
     }
-    public static void execute(Exchange exchange, MaptTemplateBean operationConfig, String inputFormat) throws Exception {
+
+    public static TemplateExecutor templateExecutor(Exchange exchange, MaptTemplateBean operationConfig, String inputFormat) throws Exception {
         OperationParams params = getOperationParams(exchange, operationConfig);
-        execute(params, exchange, inputFormat);
+        return templateExecutor(params, exchange, inputFormat);
+
     }
-    private static void execute(OperationParams params, Exchange exchange, String inputFormat) throws Exception {
 
-        if (validateParams(params)) {
-            TemplateExecutor templateExecutor = new TemplateExecutor();
-	    
-	    Reader reader = getReaderFromExchange(exchange, inputFormat, params.formatterFormat(), params.verboseReader());
+    private static TemplateExecutor templateExecutor(OperationParams params, Exchange exchange, String inputFormat) throws Exception {
+        // which custom template functions to use ???
```

Comment:
Move as an issue in the repo explaining current behaviour and changes to be discussed